framework/include/utils/InputParameters.h

```diff
@@ -750,19 +750,22 @@ class InputParameters : public Parameters
    */
   std::set<std::string> reservedValues(const std::string & name) const;
 
+  ///@{
   /**
    * Get/set a string representing the location (i.e. filename,linenum) in the input text for the
    * block containing parameters for this object.
    */
   std::string & blockLocation() { return _block_location; };
+  const std::string & blockLocation() const { return _block_location; };
+  ///@}
```

Comment:
Probably remove these trailing semicolons - we usually try to avoid using them for curly brace terminated blocks. (I see you were just copying the old code - but might as well fix it now.